lib/named-locks.js

```diff
@@ -120,42 +144,43 @@ module.exports.doWithLock = async function (name, options, func) {
  * @param {string} name The name of the lock to acquire.
  * @param {{ wait?: boolean, timeout?: number }} options Optional parameters.
  */
-async function getLock(name, options) {
-  options = options || {};
-  // TODO: we should try to ensure this only happens at most once per
-  // process; currently, we ensure the table exists every single time that
-  // we try to acquire a lock.
-  await sqldb.queryAsync(sql.ensure_named_locks_table, { name });
-
-  await sqldb.queryAsync(sql.ensure_named_lock_row, { name });
+async function getLock(name, options = {}) {
+  await pool.queryAsync(sql.ensure_named_lock_row, { name });
 
-  const client = await sqldb.beginTransactionAsync();
+  const client = await pool.beginTransactionAsync();
 
   let acquiredLock = false;
   try {
     if (options.wait && options.timeout > 0) {
-      await sqldb.callWithClientAsync(client, 'lock_timeout_set', [options.timeout]);
+      // SQL doesn't like us trying to use a parameterized query with
+      // `SET LOCAL ...`. So, in this very specific case, we do the
+      // parameterization ourselves using `escapeLiteral`.
+      await pool.queryWithClientAsync(
+        client,
+        `SET LOCAL lock_timeout = ${client.escapeLiteral(options.timeout.toString())}`,
+        {}
+      );
```

Comment:
Noting that we no longer have access to sprocs here (since we never set the search path to the random one where we created the sprocs. I suppose I could pass that through to here, but it was simpler to just remove the usage of the sproc entirely.